### PR TITLE
Enhance Kueue.Queue.Status test suite

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Kueue.Queue.Status report queuedWorkloads with assigned capacity
 QueuedWorkloadConditionType = "Assigned"

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

#### What type of PR is this?
/kind feature
/kind api-change

#### Which issue(s) this PR fixes:

Follow up on #5

### CODE REVIEW

**Comment by alculquicondor** [comment]
Can you work on an integration test first?

**Comment by ahg-g** [comment]
friendly ping, can we please make progress on this item?

**Comment by ArangoGutierrez** [comment]
> friendly ping, can we please make progress on this item?

Hi, sorry have been busy, will focus on @alculquicondor ask of more integration testing for this area this week

**Comment by ArangoGutierrez** [comment]
rebased

**Comment by ArangoGutierrez** [comment]
hold on #98 
/hold

**Comment by ArangoGutierrez** [comment]
> Can you work on an integration test first?

Ready for review

File: pkg/queue/manager.go [hunk 1]
```diff
@@ -105,13 +105,28 @@ func (m *Manager) DeleteQueue(q *kueue.Queue) {
...
+	jobs := make(map[string]int32)
+	queue, ok := m.queues[Key(q)]
+	if !ok {
+		return jobs, errQueueDoesntExist
 	}
-	return 0, errQueueDoesntExist
+	jobs["pending"] = int32(queue.heap.Len())
+
+	var admitted int32
+	for _, hi := range queue.heap.items {
```
Comment (ahg-g) [inline pkg/queue/manager.go:120]: An admitted workload wouldn't exist in the queue at all. We will need to list the workloads using the index and count the assigned ones (unless we somehow keep track of them in the Queue data structure).

Comment (ahg-g) [reply pkg/queue/manager.go:120]: and listing is going to have perf implications, I don't think we should do it. Tracking number of admitted workloads at the clusterQueue level will probably be good enough for now.

How about we use this PR to only update the test as discussed in the other PR?

Comment (ArangoGutierrez) [reply pkg/queue/manager.go:120]: Good point, I am a bit worried on the nested loop as well, but we do need something to query the queue it self, and not at the cluster lvl. 
users might want to submit several workloads, and watch for their progress

Comment (ahg-g) [reply pkg/queue/manager.go:120]: It is useful in both places, admins would be interested in checking the number of workloads admitted to a ClusterQueue. 

> users might want to submit several workloads, and watch for their progress

right, but only if we can do that in an efficient way

Comment (ArangoGutierrez) [reply pkg/queue/manager.go:120]: oh for admins I agree 100%.

@alculquicondor maybe you have an idea on how to query this without nesting?

Comment (ahg-g) [reply pkg/queue/manager.go:120]: I am not sure what you mean by nesting, but as I mentioned above, admitted workloads don't exist in `queue.heap.items` at all; the queue only have the workloads to be scheduled (pending)

Comment (ahg-g) [reply pkg/queue/manager.go:120]: what we need to do is perhaps organize the admitted workloads in each CQ broken by Q name, that is the most efficient way I believe.

Comment (ArangoGutierrez) [reply pkg/queue/manager.go:120]: If Q's are going to be namespaced, users need a way to fully manage their QW and Q's at their permission lvl 
never ever have to interact with CQ's

**Comment by ahg-g** [comment]
updated to WIP just so we know which PRs are actually waiting to be reviewed

**Comment by ArangoGutierrez** [comment]
Once https://github.com/kubernetes-sigs/kueue/pull/105 is in, I will retake work here

File: pkg/queue/manager.go (hunk 1 above)
Comment (ArangoGutierrez) [reply pkg/queue/manager.go]: >How about we use this PR to only update the test as discussed in the other PR?

Agree, done

Comment (ArangoGutierrez) [reply pkg/queue/manager.go]: >How about we use this PR to only update the test as discussed in the other PR?

Agree, done

**Comment by ArangoGutierrez** [comment]
This PR is broken will re-create 
/close

**Pull request closed without merging**